receiver/vcenterreceiver/metadata.yaml

```
@@ -0,0 +1,374 @@
+      value_type: int
+      aggregation: cumulative
+    attributes: []
+  vcenter.cluster.memory.limit:
+    enabled: true
+    description: The available memory of the cluster.
+    unit: By
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: []
+  vcenter.cluster.memory.effective:
+    enabled: true
+    description: The effective memory of the cluster. This value excludes memory from hosts in maintenance mode or are unresponsive.
+    unit: By
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: []
+  vcenter.cluster.memory.used:
+    enabled: true
+    description: The memory that is currently used by the cluster.
+    unit: By
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: []
+  vcenter.cluster.vm.count:
+    enabled: true
+    description: the number of virtual machines in the cluster.
+    unit: "{virtual_machines}"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: [vm_count_power_state]
+  vcenter.cluster.host.count:
+    enabled: true
+    description: The number of hosts in the cluster.
+    unit: "{hosts}"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: [host_effective]
+
+  vcenter.datastore.disk.usage:
+    enabled: true
+    description: The amount of space in the datastore.
+    unit: By
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: [disk_state]
+  vcenter.datastore.disk.utilization:
+    enabled: true
+    description: The utilization of the datastore.
+    unit: "%"
+    gauge:
+      value_type: double
+    attributes: []
+  vcenter.host.cpu.utilization:
+    enabled: true
+    description: The CPU utilization of the host system.
+    unit: "%"
+    gauge:
+      value_type: double
+    attributes: []
+  vcenter.host.cpu.usage:
+    enabled: true
+    description: The amount of CPU in Hz used by the host.
+    unit: "MHz"
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: []
+  vcenter.host.disk.throughput:
+    enabled: true
+    description: The throughput to the host system's disk.
+    unit: By/s
+    sum:
+      monotonic: false
+      value_type: int
+      aggregation: cumulative
+    attributes: []
+  vcenter.host.disk.latency.avg:
+    enabled: true
+    description: The latency of operations to the host system's disk.
+    unit: ms
+    gauge:
+      value_type: int
+    attributes: [latency_direction, latency_type]
+    extended_documentation: This latency is the sum of the device and kernel write latencies. Requires Performance Counter level 2 for metric to populate.
+  vcenter.host.disk.latency.total:
+    enabled: true
+    description: The total reported total latency (device and kernel times)
+    unit: ms
+    gauge:
```

Hmm, this might be a monotonic counter. What do you think?

---

Hmm I don't necessarily think these are going to be a monotonic because it will go up and down on an interval. I think perhaps the change here is a metric or description change?

description from VMware:
Highest latency value across all disks used by the host

```suggestion
  vcenter.host.disk.latency.max:
    enabled: true
    description: The highest reported latency (device and kernel times) across all disks used by the host
    unit: ms
    gauge:
```


look for disk.maxTotalLatency in this doc.
https://www.virten.net/2020/12/vsphere-7-0-performance-counter-description/

---

Hmm I did a little poking around into the most common use cases for these metrics and I think two counters come out as having the most blog posts and media.



Perf Counter                                                   |   Otel metric 
--------------------------------- | --------------------
disk.totalWriteLatency.average    | vcenter.host.disk.latency.avg (direction: write)
disk.totalReadLatency.average    | vcenter.host.disk.latency.avg (direction: read)
disk.maxTotalLatency.latest         | vcenter.host.disk.latency.max 


Before I was grabbing `disk.kernelReadLatency.average`  && `disk.kernelWriteLatency.average` however these are more specialized use cases so i am going to omit for the most useful and generic ones. Have any thoughts? Should keep the disk latency metrics a tad bit simpler as well.

---

Sounds good